src/main/java/org/truffleruby/language/supercall/LookupSuperMethodNode.java

```diff
@@ -67,10 +73,12 @@ protected DynamicObject metaClass(Object object) {
         return metaClassNode.executeMetaClass(object);
     }
 
-    @TruffleBoundary
```

Comment:
I had to delete boundary annotation because we have a frame in the method arguments

Reply:
We should pass the `DeclarationContext` context and not the entire frame, and keep the `@TruffleBoundary` here.